kumascript/src/api/util.js

```diff
@@ -195,6 +204,16 @@ class HTMLTool {
       if (!id) {
         id = generateUniqueID();
       }
+      if (isDt) {
+        // There’s existing code that causes heading IDs to get lowercased,
+        // but it doesn’t handle dt elements. So we lowercase the dt IDs
+        // here. And to ensure the ID uniqueness-check works as expected,
+        // we need to do the lowercasing after the knownIDs.has(id) check
+        // above, and not before. Also, to ensure the resulting ID will
+        // always be lowercased in all cases, we need to do the lowercasing
+        // after generateUniqueID() runs, and not before.
+        id = id.toLowerCase();
+      }
```

Comment:
> ...I guess we are comparing the H2 "Something" with the DT "something"?

Yeah, I guess so. I had thought my existing code was handling that, but clearly it wasn’t.

Anyway, I’ve now pushed a change that fixes it

> Maybe it would work better if we made a case-insensitive compare in the uniqueness check?

Well, what I ended up doing was, making sure that all the IDs that get used and pushed onto `knownIDs` are lowercased first.

As far as I can see, that’s not going to break anything — because we already have other existing code (in a later stage) that makes IDs for all H2-H6 and sections lowercase.

I guess going ahead and doing the lowercasing at this earlier stage in the code makes the lowercasing part of that later code redundant. I don’t think that redundancy is harmful, but I suppose we could go ahead and remove it as part of this PR if we wanted.